### PR TITLE
MGDAPI-236

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
implement priorityClass by adding priorityClassName managed-service-priority to the statefulSet pod spec

### CODE REVIEW

**Comment by pmccarthy** [comment]
@CathalOConnorRH @obrienrobert can this be merged?

**Comment by obrienrobert** [comment]
> @CathalOConnorRH @obrienrobert can this be merged?

@pmccarthy I discussed this with @pb82 earlier today. Before it can be merged the Grafana Operator needs to be updated to 3.6.0 and the priority class name needs to be added to the CR.

**Comment by pb82** [comment]
@CathalOConnorRH @briangallagher the AMO does currently not work when installing it outside of RHMI. The reason is that the CRDs use `Status` as a subresource, but the controllers do not use `.Status().Update()`: https://github.com/integr8ly/application-monitoring-operator/blob/master/deploy/crds/applicationmonitoring.integreatly.org_applicationmonitorings_crd.yaml#L13 vs https://github.com/integr8ly/application-monitoring-operator/blob/420e778a34a42ac3aa7ec5182da30e86bd07fa97/pkg/controller/applicationmonitoring/applicationmonitoring_controller.go#L762

I think we should fix this, maybe not in this PR but in general as there are Teams that still use the AMO to deploy a monitoring stack.

File: templates/alertmanager.yaml [hunk 1]
```diff
@@ -49,6 +49,7 @@ spec:
   secrets:
     - alertmanager-k8s-tls
     - alertmanager-k8s-proxy
+  priorityClassName: "rhoam-pod-priority"
```
Comment (obrienrobert) [inline templates/alertmanager.yaml:52]: Should this the same as what's in the `prometheus.yaml` file? E.g:
```yaml
{{- if .PriorityClassName}}
  priorityClassName: {{.PriorityClassName}}
{{- end}}
```

Comment (CathalOConnorRH) [reply templates/alertmanager.yaml:52]: Yeah, good catch, Updated now.

File: templates/alertmanager.yaml [hunk 2]
```diff
@@ -49,6 +49,9 @@ spec:
   secrets:
     - alertmanager-k8s-tls
     - alertmanager-k8s-proxy
+{{ - if .PriorityClassName }}
+  priorityClassName: {{.PriorityClassName}}
+{{ - end }}
```
Comment (obrienrobert) [inline templates/alertmanager.yaml:54]: The `-` doesn't like white space after the bracket.
```
{{- if .PriorityClassName }}
  priorityClassName: {{.PriorityClassName}}
{{- end }}
```

File: deploy/examples/ApplicationMonitoring.yaml [hunk 3]
```diff
@@ -11,3 +11,5 @@ spec:
   prometheusInstanceNamespaces: application-monitoring
   alertmanagerInstanceNamespaces: application-monitoring
   selfSignedCerts: true
+  priorityClassName: "managed-service-priority"
```
Comment (obrienrobert) [inline deploy/examples/ApplicationMonitoring.yaml:14]: Is this correct, or should it be `rhoam-pod-priority` maybe?

Comment (CathalOConnorRH) [reply deploy/examples/ApplicationMonitoring.yaml:14]: yes, I thought I had updated that.

**Comment by obrienrobert** [comment]
Verified the `priorityClassName` is added successfully to both `statefulSets`:
pod_priority_alertmanager

**Review by pb82 — APPROVED** [review]

**Pull request merged**